spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/config/ConfigServerHealthIndicator.java

```diff
@@ -53,6 +54,11 @@ protected void doHealthCheck(Health.Builder builder) throws Exception {
 				HashMap<String, Object> detail = new HashMap<>();
 				detail.put("name", environment.getName());
 				detail.put("label", environment.getLabel());
+
+				if (environment.getCached()) {
+					builder.outOfService();
```

Comment:
You are right. Regarding to the documentation out-of-service means "Status indicating that the component or subsystem has been taken out of service and should not be used.".

But down and unknown also don't fit well. Maybe unknown. Should we use a custom status like outdated or cached?